lms/lmsweb/views.py

```diff
@@ -405,8 +434,16 @@ def view(solution_id: int, file_id: Optional[int] = None):
         'role': current_user.role.name.lower(),
         'versions': versions,
         'test_results': test_results,
+        'is_shared': is_shared,
     }
 
+    if solution.is_shared:
+        view_params = {
+            **view_params,
+            'shareable_by_user': solution.is_shared,
```

Comment:
solution is already passed in view_params, there is no need to pass it again.

Reply:
Are you sure the model_to_dicts of this model will give us the is_shared?
I tried putting it in the html and I didn't see the functions in the dict - just the attributes